packages/cubejs-schema-compiler/src/compiler/CubeValidator.ts

```diff
@@ -1155,6 +1160,45 @@ export class CubeValidator implements CompilerInterface {
     return result;
   }
 
+  /**
+   * Validates that each leaf node (last segment of joinPath) appears only once across all cubes in a view.
+   * This ensures that there is only one path to reach each cube in the view's join graph.
+   */
```

Comment:
@KSDaemon Would it catch a case with these join paths?
- `a.b.c`
- `a.x.b`

Reply:
yes. Please, have a look at the tests below.